File: Shell-Sort/js/simple.js
const arr = []

function initData(count){
    for(let i = 0 ; i < count ; i++){
        arr.push(Math.floor(Math.random()*50-20))
    }
}

initData(10)

console.log(arr)


function sort(){
    for(let gap = Math.floor(arr.length/2) ; gap > 0 ; gap = Math.floor(gap/2)){
        for(let i = gap ; i < arr.length ; i++){
            // for(let j = i - gap ; j >= 0 && arr[j] > arr[j+gap] ; j -= gap){
            //     [arr[j],arr[j+gap]] = [arr[j+gap],arr[j]]
            // }
            for(let j = i - gap ; j >= 0 ; j -= gap){
                if( arr[j] > arr[j+gap] ){
                    [arr[j],arr[j+gap]] = [arr[j+gap],arr[j]]
                }
            }
        }
    }
}

sort()

console.log(arr)

File: Shell-Sort/python/simple.py
import random;

arr = [random.randint(-10,20) for _ in range(10)]

print(arr)

def sort():
    gap = int(len(arr)/2);

    while gap > 0:
        for i in range(gap,len(arr)):
            for j in range(i-gap,-1,-gap):
                if arr[j] > arr[j+gap] :
                    arr[j],arr[j+gap] = arr[j+gap],arr[j]

        gap = int(gap/2);


sort()

print(arr)
